api/src/main/java/org/hawkular/apm/api/services/Criteria.java

```diff
@@ -58,6 +58,33 @@
     private int maxResponseSize = DEFAULT_RESPONSE_SIZE;
 
     /**
+     * Default constructor.
+     */
+    public Criteria() {
+    }
+
+    /**
+     * Copy constructor.
+     *
+     * @param criteria The criteria to copy
+     */
+    public Criteria(Criteria criteria) {
+        this.startTime = criteria.startTime;
+        this.endTime = criteria.endTime;
+        this.businessTransaction = criteria.businessTransaction;
+        this.hostName = criteria.hostName;
+        this.upperBound = criteria.upperBound;
+        this.lowerBound = criteria.lowerBound;
+        this.principal = criteria.principal;
+        this.uri = criteria.uri;
+        this.operation = criteria.operation;
+
+        criteria.properties.forEach(pc -> this.properties.add(new PropertyCriteria(pc)));
```

Comment:
no because that would just share the same contents rather than have a fresh copy.